src/machine.rs

```diff
@@ -224,6 +261,20 @@ impl<'mir, 'tcx> MiriEvalContextExt<'mir, 'tcx> for MiriEvalContext<'mir, 'tcx>
     }
 }
 
+impl<'mir, 'tcx> EvalContextExt<'mir, 'tcx> for MiriEvalContext<'mir, 'tcx> {}
+/// Provides convenience methods for use elsewhere
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
+    fn i32_layout(&self) -> InterpResult<'tcx, TyAndLayout<'tcx>> {
```

Comment:
I am somewhat doubtful that these wrappers here are worth the boilerplate (assuming that the cache is going to grow over time).

Reply:
In fact, minimal boilerplate would be achieved by not making the cache lazy, and just eagerly computing the few layouts that we have here. Do you think that would make sense?

Reply:
bc54c762 switches to computing the layouts ahead of time. I had to thread a LayoutCx through to a constructor, and then unwrap an error with `Result::expect` on the other end, because I wanted to have the layouts before creating the `Machine` and, in turn, the `InterpCx`. Unwrapping an error here ought to be unreachable, because the two error enum variants are for types with unknown layouts, and types with sizes too big for the platform's `isize` to index within. Let me know what you think of how I organized this.

Reply:
Yes that looks great, thanks. :)